File: backend/app/schemas/dataset.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class DatasetCreate(BaseModel):
    name: str
    description: str | None = None
    data_type: str = "image"


class DatasetUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    status: str | None = None


class DatasetResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    data_type: str
    storage_path: str | None
    num_classes: int
    train_count: int
    val_count: int
    test_count: int
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ImageResponse(BaseModel):
    id: uuid.UUID
    dataset_id: uuid.UUID
    filename: str
    file_path: str
    width: int
    height: int
    split: str
    annotation_status: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/dataset.ts
export interface Dataset {
  id: string
  name: string
  description: string | null
  data_type: string
  storage_path: string | null
  num_classes: number
  train_count: number
  val_count: number
  test_count: number
  status: string
  created_at: string
  updated_at: string
}

export interface DatasetImage {
  id: string
  dataset_id: string
  filename: string
  file_path: string
  width: number
  height: number
  split: string
  annotation_status: string
  created_at: string
}
